### PR TITLE
Add Android project

### PR DESCRIPTION
**Pull request by marijnvdwerf**

### CODE REVIEW

**Comment by janisozaur** [comment]
Nice. :+1:

**Comment by janisozaur** [comment]
functionality of `distribution/android/setup.php` should really be merged into the gradle file

File: src/addresses.c [hunk 1]
```diff
@@ -48,6 +48,8 @@ int DISABLE_OPT RCT2_CALLPROC_X(int address, int _eax, int _ebx, int _ecx, int _
 		/* Load result with flags */
 		mov result, eax
 	}
+    #elif defined(__ANDROID__) && !defined(__i386__)
```
Comment (janisozaur) [inline src/addresses.c]: I think this should be handled with 288d11a1961270615d2657039bfbcfac677d04ed

**Comment by marijnvdwerf** [comment]
https://github.com/michel-kraemer/gradle-download-task looks like it might be useful for it.

File: src/network/twitch.cpp [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
     extern "C"
     {
+        bool gTwitchEnable = false;
```
Comment (janisozaur) [inline src/network/twitch.cpp]: ?

File: src/platform/android.c [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        return (0);
+    }
+    rval = mbrtowc(pwc, s, n, &mbs);
+    if (rval == (size_t) -1 || rval == (size_t) -2)
+        return (-1);
+    return ((int) rval);
+}
+
+bool platform_check_steam_overlay_attached() {
+    STUB();
```
Comment (janisozaur) [inline src/platform/android.c]: Given how there's no steam for android, I'd say this can be un-stubbed

File: src/network/twitch.cpp (hunk 2 above)
Comment (marijnvdwerf) [inline src/network/twitch.cpp]: At time of writing, this was causing errors when built without Twitch

Comment (IntelOrca) [inline src/network/twitch.cpp]: That's probably a bug that needs fixing.

File: src/openrct2.c [hunk 4]
```diff
@@ -332,6 +334,8 @@ void openrct2_dispose()
 	language_close_all();
 	rct2_dispose();
 	config_release();
+	munmap(segments, 16941056);
+	close(fdData);
```
Comment (IntelOrca) [inline src/openrct2.c]: This needs to be unix only - it doesn't compile on Windows.

**Comment by mrtnptrs** [comment]
Porting OpenRCT2 to Android should now be much easier with the new OpenGL renderengine, when it is finished and bug-free. Am I right?

**Comment by janisozaur** [comment]
Android does not support OpenGL

**Comment by mrtnptrs** [comment]
I thought that OpenGL ES was a bit similiar to OpenGL, but sorry, I was wrong :)

**Comment by janisozaur** [comment]
They *can be* similar, but that requires explicit support on our side. We're not doing this, at least not yet.

File: distribution/android/app/build.gradle [hunk 5]
```diff
@@ -100,3 +130,14 @@ tasks.whenTaskAdded { mergeAssetsTask ->
     mergeAssetsTask.dependsOn copyLanguageAssets
     mergeAssetsTask.dependsOn copySegmentFile
 }
+
+project.afterEvaluate {
+    // Download the licenses when (actually: just before) the
```
Comment (AaronVanGeffen) [inline distribution/android/app/build.gradle]: licenses -> libraries?

Comment (marijnvdwerf) [inline distribution/android/app/build.gradle]: Ah yeah. It's all quite a mess at the moment. Would be better to wait with your checks until it actually compiles and cleaned up the other files.

**Comment by marijnvdwerf** [comment]
Important issues:
- `g2.dat` is not being built.
- Currently uses debug builds. Release builds will need to be signed by the build server.

Other:
- No icon
- App crashes on the first run, as it doesn't have the required permissions, should probably be solved by adding another activity that acts as a splash screen.
- `openrct2_load`, languages and `g2.dat` are currently copied to the filesystem at each run. Should preferably be read from inside the `.apk`

**Comment by mrtnptrs** [comment]
@marijnvdwerf Say it when you have a build that can start up; I have a Nexus 5x with the latest Android N/Nougat/7 preview version. So, I can check the compatibility with the newest Android version and do more testing for you.

**Comment by mrtnptrs** [comment]
Just curious, why did you close this? By accident or did it not work or something?

**Pull request closed without merging**